### ISSUE TITLE
What is an "unquoted string"?

### ISSUE THREAD

**Issue by ChrisJefferson** [arangodb/arangodb#3666]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x ] 3.2

On https://docs.arangodb.com/3.2/AQL/Fundamentals/DataTypes.html, attribute names must be a "quoted or unquoted string".

I've been asked to extend a JSON parser to support this format, but I don't know what arangodb counts as a valid unquoted string.

**Comment by jsteemann** [comment]
A quoted string is a string enclosed in double or single quotes, e.g. `"foo"`, `"foo bar"`, `"foo\nbar"`, `'foo'.
An unquoted string is a string without the surrounding quotes. In order to make parsing unambiguous, there are some restrictions for unquoted strings:
1) they must not contain any whitespace
1) nor escape sequences
1) they are restricted to a specific set of characters, i.e. the ASCII letters, the digits 0-9, underscore and the dollar sign
1) must not be AQL keywords
In short, the first 3 restrictions translate into the (flex) regular expression `($?[a-zA-Z][_a-zA-Z0-9]*|_+[a-zA-Z]+[_a-zA-Z0-9]*)`.
The latter restriction means if an AQL keyword is used as an unquoted string, it will result in a parse error.

**Comment by Simran-B** [comment]
> I've been asked to extend a JSON parser to support this format

Is that parser supposed to process query results? You won't find unquoted strings there, as the output is strict JSON. Unquoted strings only exist inside of AQL queries.

**Comment by dothebart** [comment]
I will close this question as solved.

**Comment by uluQulu** [comment]
thanks @jsteemann for explanation 🙌🏼 that was what I was looking for 👌🏼

**Comment by Simran-B** [comment]
Should be clearer in the docs with these changes:
- 3.5 https://github.com/arangodb/arangodb/pull/8287/
- 3.4 https://github.com/arangodb/arangodb/pull/8288/
